src/ng/location.js

```diff
@@ -748,8 +748,10 @@ function $LocationProvider() {
    *     whether or not a <base> tag is required to be present. If `enabled` and `requireBase` are
    *     true, and a base tag is not present, an error will be thrown when `$location` is injected.
    *     See the {@link guide/$location $location guide for more information}
-   *   - **rewriteLinks** - `{boolean}` - (default: `true`) When html5Mode is enabled,
-   *     enables/disables url rewriting for relative links.
+   *   - **rewriteLinks** - `{boolean|String}` - (default: `true`) When html5Mode is enabled,
+   *     enables/disables url rewriting for relative links. If set to a string, url rewriting will
+   *     only happen on links with an attribute that matches the given string. For example, if set
+   *     to "internalLink", then the URL will only be rewritten for `<a internal-link>` links.
```

Comment:
There is no attribute name (de)normalization going on. In order to match `<a internal-link>` you need to set `rewriteLinks` to `'internal-link'` (not `'internalLink'`).